code/game/mecha/mecha_topic.dm

```diff
@@ -306,6 +307,24 @@
 		maint_access = !maint_access
 		send_byjax(src.occupant,"exosuit.browser","t_maint_access","[maint_access?"Forbid":"Permit"] maintenance protocols")
 
+	if (href_list["toggle_port_connection"])
+		if(internal_tank.connected_port)
+			if(internal_tank.disconnect())
+				occupant_message("Disconnected from the air system port.")
+				log_message("Disconnected from gas port.")
+			else
+				occupant_message("<span class='warning'>Unable to disconnect from the air system port!</span>")
+				return
+		else
+			var/obj/machinery/atmospherics/components/unary/portables_connector/possible_port = locate() in loc
+			if(internal_tank.connect(possible_port))
```

Comment:
possible_port may be null, does connect handle that?

Reply:
@Cyberboss Yes

`portable_atmospherics.dm`
```
/obj/machinery/portable_atmospherics/proc/connect(
    obj/machinery/atmospherics/components/unary/portables_connector/new_port)
    if(connected_port || !new_port || new_port.connected_device)
		return 0
```